packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesController.kt

```diff
@@ -35,10 +37,23 @@ class UpdatesController {
           null
         }
 
-        singletonInstance = if (UpdatesConfiguration.canCreateValidConfiguration(context, configuration) && updatesDirectory != null) {
+        val canCreateValidConfiguration = UpdatesConfiguration.canCreateValidConfiguration(context, configuration)
+        singletonInstance = if (canCreateValidConfiguration && updatesDirectory != null) {
           val updatesConfiguration = UpdatesConfiguration(context, null)
           EnabledUpdatesController(context, updatesConfiguration, updatesDirectory)
         } else {
+          val logger = UpdatesLogger(context)
+          if (!canCreateValidConfiguration) {
+            logger.warn(
+              "The expo-updates system is disabled, either explicitly or by invalid configuration. If you wish to enable it, ensure a valid URL and runtime version are supplied.",
+              UpdatesErrorCode.InitializationError
+            )
+          } else {
+            logger.error(
+              "The expo-updates system is disabled due to a storage access error: ${updatesDirectoryException?.message ?: "Unknown Error"}",
+              UpdatesErrorCode.InitializationError
+            )
```

Comment:
I'm a little worried about unexpected non-storage access errors leading to this warning. Perhaps this would help?

```suggestion
else if (updatesDirectoryException != null) {
  `The expo-updates system is disabled due to a storage access error: ${updatesDirectoryException.message}`
} else {
  `The expo-updates system is disabled due to an unexpected error <ideally more debugging info here>`
}
```